src/Config.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE TypeApplications #-}
+
+module Config (readConfig, peopleAndGroups) where
+
+import Data.Text (Text, isInfixOf, pack, toLower)
+import qualified Data.Text.IO as T
+import Data.Yaml (FromJSON (..), (.:))
+import qualified Data.Yaml as Y
+import Entities (Person (..))
+
+data Group = Group
+  { groupName :: Text,
+    groupEmails :: [Text]
+  }
+  deriving (Eq, Show)
+
+instance FromJSON Group where
+  parseJSON = Y.withObject "Group" $ \v ->
+    Group
+      <$> v .: "name"
+      <*> v .: "emails"
+
+newtype Config = Config
+  { groups :: [Group]
+  }
+  deriving (Eq, Show)
+
+instance FromJSON Config where
+  parseJSON = Y.withObject "Config" $ \v ->
+    Config <$> v .: "groups"
+
+readConfig :: FilePath -> IO Config
+readConfig cp = do
+  result <- Y.decodeFileEither @Config cp
+  case result of
+    Left e -> do
+      T.putStrLn $ "Error parsing config: " <> pack (show e)
+      -- Return default (empty for now) config if parsing fails.
+      return Config {groups = []}
+    Right config -> return config
```

Comment:
I'm not convinced this should continue after failing to parse config — thoughts?